### PR TITLE
Docs usage section

### PR DESCRIPTION
**Pull request by asvetlov**
A review is welcome!

### CODE REVIEW

File: docs/jobs_usage.rst [hunk 1]
```diff
@@ -1,23 +1,172 @@
...
+   await client.jobs.kill(job.id)
 
 
+Expose job's TCP ports locally
+==============================
+
+Sometimes there is a need to access TCP endpoints exposed by a job executed on the
+Neuromation Platform from local workstation.
+
+For example, you've started a gRPC server inside a container on TCP port ``12345`` and
```
Comment (asvetlov) [inline docs/jobs_usage.rst]: I'd like to provide a real (or real-like) example here for binary TCP protocol (HTTP port can be exposed by `container.http.port` property without port forwarding.
Ideas?

**Comment by asvetlov** [comment]
Not sure if `images` and `users` should be described in *usage* section. At least it is not the first priority.
I expect that most people will use CLI for image pushing and permissions granting.

File: docs/jobs_usage.rst [hunk 2]
```diff
@@ -1,23 +1,172 @@
...
+       storage_uri=URL("storage:folder"),
+       container_path="/var/data",
+   )
+
+   job = await client.jobs.run(
+       Container(
+           image=client.parse.remote_image("ubuntu:latest"),
+           resources=Resources(memory_mb=100, cpu=0.5),
+           command="sleep 30",
+           volumes=[volume],
```
Comment (shagren) [inline docs/jobs_usage.rst]: Read-only volume example with path  ` storage://neuromation/public` will be helpful too.

File: docs/storage_usage.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+   async with get() as client:
+       await client.storage.upload_dir(
+           URL("file:local_folder"),
+           URL("storage:"remote_folder"),
+       )
+
+The example above recursively uploads all files and directories ``./local_folder`` to
+``storage://<username>/remote_folder``.
+
+Use ``update=True`` flag to upload only files that are newer than are present on the
```
Comment (shagren) [inline docs/storage_usage.rst]: Not present or obsoletes?

File: docs/jobs_usage.rst [hunk 4]
```diff
@@ -1,23 +1,172 @@
...
+   from grpclib.client import Channel
+
+   # generated by protoc
+   from .helloworld_pb2 import HelloRequest, HelloReply
+   from .helloworld_grpc import GreeterStub
+
+    async with client.jobs.port_forward(job.id, 23456, 12345):
+        # open gRPC client and use it
+
+        channel = Channel('127.0.0.1', 50051)
```
Comment (dalazx) [inline docs/jobs_usage.rst]:
```suggestion
        channel = Channel('127.0.0.1', 23456)
```

File: docs/jobs_usage.rst [hunk 5]
```diff
@@ -1,23 +1,172 @@
...
+
 .. _job-preemption:
 
 Job preemption
 ==============
 
-TBD
+Job preemption means that unlike normal jobs preemptible ones can be stopped by kernel
+when the system has lack of resources and restarted later.  All memory changes are lost
+but disk data is persistent.
```
Comment (dalazx) [inline docs/jobs_usage.rst]: should we point out that by persistent disk here we mean specifically the Neuromation Storage mounts? the host disks are lost.

File: docs/storage_usage.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+Use :meth:`Storage.upload_dir` to upload a local directory on the Neuromation Storage::
+
+   from neuromation.api import *
+   from yarl import URL
+
+   async with get() as client:
+       await client.storage.upload_dir(
+           URL("file:local_folder"),
+           URL("storage:"remote_folder"),
```
Comment (dalazx) [inline docs/storage_usage.rst]:
```suggestion
           URL("storage:remote_folder"),
```

File: docs/storage_usage.rst [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+The example above recursively uploads all files and directories ``./local_folder`` to
+``storage://<username>/remote_folder``.
+
+Use ``update=True`` flag to upload only files that are newer than are present on the
+Storage::
+
+   await client.storage.upload_dir(
+       URL("file:local_folder"),
+       URL("storage:"remote_folder"),
```
Comment (dalazx) [inline docs/storage_usage.rst]:
```suggestion
       URL("storage:remote_folder"),
```

File: docs/storage_usage.rst [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+Download a Folder
+=================
+
+Use :meth:`Storage.download_dir` for downloading data from the Neuromation Storage to
+local disk.
+
+The method is a counterpart to :meth:`Storage.upload_dir` and has the same arguments::
+
+   await client.storage.download_dir(
+       URL("storage:"remote_folder"),
```
Comment (dalazx) [inline docs/storage_usage.rst]:
```suggestion
       URL("storage:remote_folder"),
```

File: docs/storage_usage.rst [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+       URL("file:local_folder"),
+   )
+
+The example above recursively downloads files and directories from
+``storage:"remote_folder`` to ``./local_folder``.
```
Comment (dalazx) [inline docs/storage_usage.rst]:
```suggestion
``storage:remote_folder`` to ``./local_folder``.
```

File: docs/jobs_usage.rst (hunk 2 above)
Comment (asvetlov) [reply docs/jobs_usage.rst]: fixed

File: docs/jobs_usage.rst (hunk 5 above)
Comment (asvetlov) [reply docs/jobs_usage.rst]: the doc is updated

**Pull request merged**